tests/net/lib/coap_client/src/main.c

```diff
@@ -676,9 +667,8 @@ ZTEST(coap_client, test_unmatching_tokens)
 	zassert_true(ret >= 0, "Sending request failed, %d", ret);
 	set_socket_events(ZSOCK_POLLIN);
 
-	k_sleep(K_MSEC(1));
-	k_sleep(K_MSEC(1));
+	k_sleep(K_MSEC(2));
 	clear_socket_events();
-	k_sleep(K_MSEC(500));
+	k_sleep(K_MSEC(700));
```

Comment:
I have used these kind of sleeps myself in the past, but as the tests run by PR already take long time, these sleeps just add up and make CI runs longer. So it would make sense to try to either lower the sleeps or handle this differently. Anyway, this does not need to be handled in this PR, just a suggestion for future work.